Source/WTF/wtf/glib/GWeakPtr.h

```diff
@@ -0,0 +1,111 @@
+/*
+ *  Copyright (C) 2023 Igalia S.L.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Library General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Library General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Library General Public License
+ *  along with this library; see the file COPYING.LIB.  If not, write to
+ *  the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ *  Boston, MA 02110-1301, USA.
+ *
+ */
+
+#pragma once
+
+#if USE(GLIB)
+
+#include <glib-object.h>
+#include <wtf/Noncopyable.h>
+
+namespace WTF {
+
+template <typename T> class GWeakPtr {
+    WTF_MAKE_NONCOPYABLE(GWeakPtr);
+public:
+    GWeakPtr() = default;
+
+    explicit GWeakPtr(T* ptr)
+        : m_ptr(ptr)
+    {
+        RELEASE_ASSERT(!ptr || G_IS_OBJECT(ptr));
+        addWeakPtr();
+    }
+
+    GWeakPtr(GWeakPtr&& other)
+    {
+        auto* ptr = other.get();
+        reset(ptr);
```

Comment:
I would have used a single line here, but no strong preference:

```suggestion
        reset(other.get());
```

Reply:
Oh yeah, I noticed that too but just forgot to point it out. :P